modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -138,18 +154,54 @@ SynonymMap buildSynonyms(Analyzer analyzer, ReaderWithOrigin rules) {
     }
 
     protected ReaderWithOrigin getRulesFromSettings(Environment env) {
+
         if (settings.getAsList("synonyms", null) != null) {
             List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
             StringBuilder sb = new StringBuilder();
             for (String line : rulesList) {
                 sb.append(line).append(System.lineSeparator());
             }
             return new ReaderWithOrigin(new StringReader(sb.toString()), "'" + name() + "' analyzer settings");
+        } else if ((settings.get("synonyms_set") != null) && SynonymsAPI.isEnabled()) {
+            if (analysisMode != AnalysisMode.SEARCH_TIME) {
```

Comment:
Maybe we can refactor this into a separate method for retrieving the `ReaderWithOrigin` and make clearer the case distinction for the different synonyms origin.

Reply:
Thanks for the feedback, addressed in 334caf6cc4ff8919a797b4688a28b5bbd824e945